test/e2e/network/endpointslice.go

```diff
@@ -0,0 +1,428 @@
+	version := "v1"
+	ginkgo.Context("version "+version, func() {
+		f := framework.NewDefaultFramework("endpointslice")
+
+		var cs clientset.Interface
+		var podClient *framework.PodClient
+
+		ginkgo.BeforeEach(func() {
+			cs = f.ClientSet
+			podClient = f.PodClient()
+		})
+
+		labelPod1 := "pod1"
+		labelPod2 := "pod2"
+		labelPod3 := "pod3"
+		labelShared12 := "shared12"
+		labelValue := "on"
+
+		ginkgo.It("should create Endpoints and EndpointSlices for Pods matching a Service", func() {
+			svc1 := createServiceReportErr(cs, f.Namespace.Name, &v1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "example-int-port",
+				},
+				Spec: v1.ServiceSpec{
+					Selector: map[string]string{labelPod1: labelValue},
+					Ports: []v1.ServicePort{{
+						Name:       "example",
+						Port:       80,
+						TargetPort: intstr.FromInt(3000),
+						Protocol:   v1.ProtocolTCP,
+					}},
+				},
+			})
+
+			svc2 := createServiceReportErr(cs, f.Namespace.Name, &v1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "example-named-port",
+				},
+				Spec: v1.ServiceSpec{
+					Selector: map[string]string{labelShared12: labelValue},
+					Ports: []v1.ServicePort{{
+						Name:       "http",
+						Port:       80,
+						TargetPort: intstr.FromString("example-name"),
+						Protocol:   v1.ProtocolTCP,
+					}},
+				},
+			})
+
+			svc3 := createServiceReportErr(cs, f.Namespace.Name, &v1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "example-no-match",
+				},
+				Spec: v1.ServiceSpec{
+					Selector: map[string]string{labelPod3: labelValue},
+					Ports: []v1.ServicePort{{
+						Name:       "example-no-match",
+						Port:       80,
+						TargetPort: intstr.FromInt(8080),
+						Protocol:   v1.ProtocolTCP,
+					}},
+				},
+			})
+
+			pod1 := podClient.Create(&v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "pod1",
+					Labels: map[string]string{
+						labelPod1:     labelValue,
+						labelShared12: labelValue,
+					},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  "container1",
+							Image: imageutils.GetE2EImage(imageutils.Nginx),
+							Ports: []v1.ContainerPort{{
+								Name:          "example-name",
+								ContainerPort: int32(3000),
+							}},
+						},
+					},
+				},
+			})
+
+			pod2 := podClient.Create(&v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "pod2",
+					Labels: map[string]string{
+						labelPod2:     labelValue,
+						labelShared12: labelValue,
+					},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  "container1",
+							Image: imageutils.GetE2EImage(imageutils.Nginx),
+							Ports: []v1.ContainerPort{{
+								Name:          "example-name",
+								ContainerPort: int32(3001),
+							}, {
+								Name:          "other-port",
+								ContainerPort: int32(3002),
+							}},
+						},
+					},
+				},
+			})
+
+			pod1 = waitForPodWithIP(cs, f.Namespace.Name, pod1.Name)
```

Comment:
consider moving this into expectEndpointsAndSlices

Reply:
I ended up leaving it where it is since those functions are relying on the same set of pods existing, whereas they are each looking for unique sets of Endpoints and EndpointSlices. Hopefully with how things have been restructured now the separation makes a bit more sense, but happy to move things around if not.